packages/functional-tests/pages/products/index.ts

```diff
@@ -11,6 +11,10 @@ export class SubscribePage extends BaseLayout {
     return this.page.getByRole('heading', { name: 'Promo Code Applied' });
   }
 
+  async getCouponStatusByDataTestId(dataTestId: string) {
+    return this.page.locator(`[data-testid="${dataTestId}"]`);
+  }
```

Comment:
Playwright recommends using [built-in locators](https://playwright.dev/docs/locators), applying that recommendation + standardizing the use of accessors in page objects would look like:

```
  get couponError() {
    return this.page.getByTestId('coupon-error');
  }

  get removeCouponButton() {
    return this.page.getByTestId('coupon-remove-button');
  }
```